### DISCUSSION TITLE
Postgres Extensions

### DISCUSSION THREAD

**Discussion by nunopato** [nhost/nhost#1020] [General]
This discussion is to keep track of requests for Postgres Extensions. Please comment with a new extension (if not present) or give a thumbs up to existing ones.

**Comment by nunopato** [comment]
pg_cron (https://github.com/citusdata/pg_cron)

**Reply by pkreipke** [reply]
Doesn't Hasura already have something like this? Perhaps only at GQL level I guess but still.

**Reply by nunopato** [reply]
@pkreipke yes, Hasura cron triggers are used to trigger HTTP endpoints (with custom logic) periodically. pg_cron allows you to schedule SQL statements directly on the database, so there are different use cases.

**Reply by quentin-decre** [reply]
Really really looking forward to have pg_cron :D

**Comment by nunopato** [comment]
pgRouting (https://pgrouting.org/)

**Comment by nunopato** [comment]
pgAudit (https://www.pgaudit.org/)

**Comment by Svarto** [comment]
Apache AGE® is a PostgreSQL extension that provides graph database functionality.

https://age.apache.org/

**Comment by Svarto** [comment]
Making Postgres and Elasticsearch work together like it's 2022

https://www.zombodb.com/

**Comment by yureckey** [comment]
PLV8
"PLV8 is a trusted Javascript language extension for PostgreSQL. It can be used for stored procedures, triggers, etc"

**Comment by shiftlabs1** [comment]
postgres http plugin https://github.com/pramsey/pgsql-http
Motivation: Most times you want an hasura event trigger to call an external API Method with data from multiple tables not just the triggering table. from a postgres trigger function , one can aggregate the data and make the call using this module. It's first class in the likes of supabase

**Comment by pkreipke** [comment]
Wow. So neat, didn't know about many of these!

Just curious though: if we don't use any of the extensions but NHost makes them available - is that going to be opt-in? Does having extensions enabled but not being used impact PG performance? E.g. right now we don't have any need for PG GIS etc.

**Reply by nunopato** [reply]
@pkreipke they will always be opt-in. Having extensions available should only increase the image size, but not have any significant impact in performance. Having them available in the image is not the same as enabling/installing them with `create extension an-extension`.

**Reply by pkreipke** [reply]
Got it thanks! Extensions not something I've needed before. Looking forward to possibilities!

**Comment by Weijtmans** [comment]
I would love to see support for [TimescaleDB](https://www.timescale.com/); an extension that optimizes Postgres for time-series data.

**Reply by anwfr** [reply]
I'd love support for TimescaleDB.

**Comment by spakanati** [comment]
If this is still a right place to discuss extensions, I would love to see `pg_hashids` like Supabase has: https://docs-fe4unoip4-supabase.vercel.app/docs/guides/database/extensions/pg_hashids

**Reply by quentin-decre** [reply]
Oh very interesting, I would have a great use of it too !

**Reply by dbarrosop** [reply]
Here is a good place as any. Anyway, good timing as I was adding another extension and updating the existing one so I just added this one too:

https://hub.docker.com/repository/docker/nhost/postgres/general

I will update the docs and the recommended versions in the dashboard during the week but the image is ready to be used if you want to try it out.

**Comment by wildsurfer** [comment]
pgRouting (https://pgrouting.org/)

**Reply by dbarrosop** [reply]
https://github.com/nhost/nhost/pull/3820

**Comment by vpamrit** [comment]
Any chance we can update the `pg_search` (ParadeDB) extension? 

Unclear if this is an issue with my installation or the way the pg_search extension is configured to install but:
1) all functions are installed under the `paradedb` namespace instead of `pdb` (which is incongruous with the documentation)
2) tokenizers are not installed (ngram, icu, etc.)
3) version is a bit outdated `0.17.2` (at least on my local setup!)

Sincere apologies if I am missing something obvious here or if there's a way I can directly manage/update the extension :)

**Reply by quentin-decre** [reply]
They are going to do some improvements on joins in the next month. I wait a little bit more. 

Current version is more stable than 17.2

**Comment by dbarrosop** [comment]
I am locking this conversation, let's use regular github issues to request/track extensions.
